pkg/graveler/graveler.go

```diff
@@ -1898,6 +1929,7 @@ func (g *Graveler) ListStaging(ctx context.Context, branch *Branch, batchSize in
 
 // listStagingArea Returns an iterator which is an aggregation of all changes on all the branch's staging area (staging + sealed)
 // for each key in the staging area it will return the latest update for that key (the value that appears in the newest token)
+// listStagingArea will not return changes that were already compacted and saved in the CompactedBaseMetaRangeID
```

Comment:
Note that if this stays without compaction it still has some usages that expect it to be all of staging (i.e with compaction). For example all of the commands that can't work on "dirty" branches, such as merge, should not work If there is a changes on compacted data and no changes in staging